### PR TITLE
[SC-210116] E2E tests for sample list filtering

### PR DESCRIPTION
**Pull request by luisczi**
### Summary:
- **What:**  This PR implements filtering of samples using various criteria
- **Ticket:** [sc-210116](https://app.shortcut.com/genepi/story/sc-210116)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/tests/utils/archive/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    if (statuses[i] == desireStatus) {
+      counter++;
+    }
+  }
+  await expect(statuses).toHaveLength(counter);
+}
+
+export async function filterLineage(page: Page, lineages: string[]) {
+  let actualLineage = 0;
+  while (actualLineage < lineages.length) {
```
Comment (charles-testco) [inline src/frontend/tests/utils/archive/helpers.ts:90]: @luisczi  I believe we don't need these functions any more.
Also, please don't update  legacy code in the archive folder - they are there for compatibility; e2e code should all be in E2E folder except very few things we want to share with unit tests

File: src/frontend/e2e/setup/global-setup.ts [hunk 2]
```diff
@@ -17,11 +18,17 @@ async function globalSetup(config: FullConfig): Promise<void> {
...
+    page.waitForEvent('domcontentloaded'),
+    page.locator('button#onetrust-accept-btn-handler').click(),
+  ])
+  await banner.waitForLoadState('domcontentloaded');
   await page.context().storageState({ path: storageState as string });
 
   const cookies = await page.context().cookies();
   const cookieString = JSON.stringify(cookies);
-  fs.writeFileSync(cookieStorage, cookieString);
+  const hey = JSON.parse(cookieString)[4].value;
```
Comment (charles-testco) [inline src/frontend/e2e/setup/global-setup.ts]: Please give a meaningful name

**Review by charles-testco — CHANGES_REQUESTED** [review]
@luisczi the changes you made in global-setup has caused all tests to fail

File: src/frontend/e2e/pages/filter.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import { Page } from "@playwright/test";
+export abstract class FilterSample {
+  public static async applyFilter(
+    page: Page,
+    filterData: Partial<FilterData>
+  ): Promise<void> {
+    // fill in upload date(s)
+    await page.waitForFunction(() => {
```
Comment (charles-testco) [inline src/frontend/e2e/pages/filter.ts:8]: Please comment so we know what is going on here

**Review by charles-testco — CHANGES_REQUESTED** [review]

File: src/frontend/e2e/setup/global-setup.ts (hunk 2 above)
Comment (luisczi) [reply src/frontend/e2e/setup/global-setup.ts]: this is fixed now

File: src/frontend/tests/utils/archive/helpers.ts (hunk 1 above)
Comment (luisczi) [reply src/frontend/tests/utils/archive/helpers.ts:90]: Screen Shot 2022-09-09 at 12 45 53 PM

i dont see that implementation within this branch, actually i havent touch **helpers.ts** file you can even see that on my commit changes

File: src/frontend/e2e/pages/filter.ts (hunk 3 above)
Comment (luisczi) [reply src/frontend/e2e/pages/filter.ts:8]: comment ready

**Comment by luisczi** [comment]
> @luisczi the changes you made in global-setup has caused all tests to fail

@charles-testco my changes that i have perform on global setup doesnt break the tests

Screen Shot 2022-09-09 at 12 29 54 PM

i will remove those lines then

File: src/frontend/tests/utils/archive/helpers.ts (hunk 1 above)
Comment (charles-testco) [reply src/frontend/tests/utils/archive/helpers.ts:90]: I have already removed it after my comment. This was the commit:
commit 0fa91ee96d1dc7562d482dd08ae27e6600b2bc0e
Author: luisczi <lbrambila@contractor.chanzuckerberg.com>
Date:   Tue Aug 9 23:41:09 2022 -0500

    started filters test


If you type "git show 0fa91ee96d1dc7562d482dd08ae27e6600b2bc0e" you will see the changes

**Review by charles-testco — APPROVED** [review]

**Pull request merged**